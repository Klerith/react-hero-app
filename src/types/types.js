

export const types = {
    login:  '[auth] login',
    logout: '[auth] logout',
}