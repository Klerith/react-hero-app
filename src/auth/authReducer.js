import { types } from '../types/types';

// const state = {
//     name: 'Fernando',
//     logged: true
// }


export const authReducer = (state = {}, action ) => {

    switch ( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }

        case types.logout:
            return {
                logged: false
            }
    
        default:
            return state;
    }

}