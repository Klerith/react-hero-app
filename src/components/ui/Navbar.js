import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" 
                to="/"
            >
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/dc"
                    >
                        DC
                    </NavLink>
                    
                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/search"
                    >
                        Search
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
                <ul className="navbar-nav ml-auto">
                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/login"
                    >
                        Logout
                    </NavLink>
                </ul>
            </div>
        </nav>
    )
}