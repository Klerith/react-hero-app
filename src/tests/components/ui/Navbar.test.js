import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router-dom';
import '@testing-library/jest-dom';

import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/Navbar';
import { types } from '../../../types/types';



describe('Pruebas en <Navbar />', () => {

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }
    
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Pedro'
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>
            <MemoryRouter>
                <Router history={ historyMock }>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('debe de mostrarse correctamente', () => {
     
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text-info').text().trim() ).toBe('Pedro');

    });

    test('debe de llamar el logout y el usar history', () => {
        
        wrapper.find('button').prop('onClick')();


        expect( contextValue.dispatch ).toHaveBeenCalledWith({
            type: types.logout
        });

        expect( historyMock.replace ).toHaveBeenCalledWith('/login');

    })
    
    

})
