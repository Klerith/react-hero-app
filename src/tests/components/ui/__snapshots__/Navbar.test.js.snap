// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <Link
    className="navbar-brand"
    to="/"
  >
    <LinkAnchor
      className="navbar-brand"
      navigate={[Function]}
    >
      <a
        className="navbar-brand"
        onClick={[Function]}
      >
        Asociaciones
      </a>
    </LinkAnchor>
  </Link>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        exact={true}
        to="/marvel"
      >
        <Link
          aria-current={null}
          className="nav-item nav-link"
          to={
            Object {
              "hash": "",
              "pathname": "/marvel",
              "search": "",
              "state": null,
            }
          }
        >
          <LinkAnchor
            aria-current={null}
            className="nav-item nav-link"
            navigate={[Function]}
          >
            <a
              aria-current={null}
              className="nav-item nav-link"
              onClick={[Function]}
            >
              Marvel
            </a>
          </LinkAnchor>
        </Link>
      </NavLink>
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        exact={true}
        to="/dc"
      >
        <Link
          aria-current={null}
          className="nav-item nav-link"
          to={
            Object {
              "hash": "",
              "pathname": "/dc",
              "search": "",
              "state": null,
            }
          }
        >
          <LinkAnchor
            aria-current={null}
            className="nav-item nav-link"
            navigate={[Function]}
          >
            <a
              aria-current={null}
              className="nav-item nav-link"
              onClick={[Function]}
            >
              DC
            </a>
          </LinkAnchor>
        </Link>
      </NavLink>
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        exact={true}
        to="/search"
      >
        <Link
          aria-current={null}
          className="nav-item nav-link"
          to={
            Object {
              "hash": "",
              "pathname": "/search",
              "search": "",
              "state": null,
            }
          }
        >
          <LinkAnchor
            aria-current={null}
            className="nav-item nav-link"
            navigate={[Function]}
          >
            <a
              aria-current={null}
              className="nav-item nav-link"
              onClick={[Function]}
            >
              Search
            </a>
          </LinkAnchor>
        </Link>
      </NavLink>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Pedro
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
